### PR TITLE
Omit current date from fact_daily_guideline_checks

### PR DESCRIPTION
**Pull request by owades**
# Description

I observed that after 12am UTC (4pm pacific), guideline check dashboard in metabase shows skewed results, as it shows incomplete data from the current day.

This PR addresses this by preventing feed_guideline_index from ever including the current date.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?

I ran this change on today's data, visible in `cal-itp-data-infra-staging.owades_mart_gtfs_guidelines.fact_daily_guideline_checks`. Verified that the most recent data is yesterday, while the main table contains data from today.

### CODE REVIEW

**Comment by owades** [comment]
@lauriemerrell I'm interested in whether you'd suggest making this change upstream or downstream from here - thanks for reviewing!

**Review by lauriemerrell — APPROVED** [review]
> @lauriemerrell I'm interested in whether you'd suggest making this change upstream or downstream from here - thanks for reviewing!

Yeah, this is kind of a tricky one. I talked about it a bit with @atvaccaro... I think this is fine, and longer term we may want to actually look into the Metabase side of this too (already exists as #1064). 

The main concern here would just be that theoretically this solution may only work as intended because of quirks in the scheduled execution times of the various jobs, but I think for this use case it seems appropriate, and we probably should put more thought into how we handle this in the underlying `fact daily feeds` and `dim feeds`.

Defining this at the index seems good because it's a single point of intervention (i.e., not doing something different for each individual check) and I think it signals "this shouldn't even be evaluated" as opposed to truncating in the mart which could be interpreted as "we've evaluated this and we're just not displaying it". 

TLDR: LGTM.

**Comment by owades** [comment]
Thank you!
